src/Features/Core/Portable/Completion/CompletionItem.cs

```diff
@@ -3,220 +3,225 @@
+        /// <param name="span">The span of the syntax element in the document associated with this item.</param>
+        /// <param name="properties">Additional information.</param>
+        /// <param name="tags">Descriptive tags that may influence how the item is displayed.</param>
+        /// <param name="rules">The rules that declare how this item should behave.</param>
+        /// <returns></returns>
+        public static CompletionItem Create(
             string displayText,
-            TextSpan filterSpan,
-            ImmutableArray<SymbolDisplayPart> description = default(ImmutableArray<SymbolDisplayPart>),
-            Glyph? glyph = null,
-            string sortText = null,
             string filterText = null,
-            bool preselect = false,
-            bool isBuilder = false,
-            bool showsWarningIcon = false,
-            bool shouldFormatOnCommit = false,
-            CompletionItemRules rules = null,
-            ImmutableArray<CompletionItemFilter>? filters = null)
-            : this(completionProvider, displayText, filterSpan,
-                   description.IsDefault ? (Func<CancellationToken, Task<ImmutableArray<SymbolDisplayPart>>>)null : c => Task.FromResult(description),
-                   glyph, /*hasAsyncDescription*/ false, sortText, filterText, preselect, isBuilder, showsWarningIcon, shouldFormatOnCommit, rules, filters)
+            string sortText = null,
+            TextSpan span = default(TextSpan),
+            ImmutableDictionary<string, string> properties = null,
+            ImmutableArray<string> tags = default(ImmutableArray<string>),
+            CompletionItemRules rules = null)
         {
+            return new CompletionItem(
+                span: span,
+                displayText: displayText,
+                filterText: filterText,
+                sortText: sortText,
+                properties: properties,
+                tags: tags,
+                rules: rules);
         }
 
-        // Constructor kept for back compat.  When we move to our new completion API we can remove this.
-        public CompletionItem(
-            CompletionListProvider completionProvider,
-            string displayText,
-            TextSpan filterSpan,
-            Func<CancellationToken, Task<ImmutableArray<SymbolDisplayPart>>> descriptionFactory,
-            Glyph? glyph,
-            string sortText,
-            string filterText,
-            bool preselect,
-            bool isBuilder,
-            bool showsWarningIcon,
-            bool shouldFormatOnCommit,
-            CompletionItemRules rules)
-            : this(completionProvider, displayText, filterSpan, descriptionFactory, glyph, sortText,
-                  filterText, preselect, isBuilder, showsWarningIcon, shouldFormatOnCommit, rules,
-                  ImmutableArray<CompletionItemFilter>.Empty)
+        private CompletionItem With(
+            Optional<TextSpan> span = default(Optional<TextSpan>),
```

Comment:
Why the "Optional" With pattern here?  But elsewhere we do the WithFoo WithBar pattern?